miniapps/tools/gridfunction-bounds.cpp

```diff
@@ -0,0 +1,294 @@
+//     functions: Applications to mesh validity and bounds-preserving limiters".
+//
+//
+// Compile with: make gridfunction-bounds
+//
+// Sample runs:
+// mpirun -np 4 gridfunction-bounds
+// mpirun -np 4 gridfunction-bounds -nb 100 -ref 5 -bt 2 -l2
+
+#include "mfem.hpp"
+#include <iostream>
+#include <fstream>
+
+using namespace mfem;
+using namespace std;
+
+void VisualizeField(ParMesh &pmesh, ParGridFunction &input,
+                    char *title, int pos_x, int pos_y);
+
+int main (int argc, char *argv[])
+{
+   // 0. Initialize MPI and HYPRE.
+   Mpi::Init(argc, argv);
+   int myid = Mpi::WorldRank();
+   int nranks = Mpi::WorldSize();
+   Hypre::Init();
+
+   // Set the method's default parameters.
+   const char *mesh_file = "../gslib/triple-pt-1.mesh";
+   const char *sltn_file = "../gslib/triple-pt-1.gf";
+   int  ref              = 2;
+   bool visualization    = true;
+   bool visit            = false;
+   int  b_type           = -1;
+   bool continuous       = true;
+   int  nbrute           = 0;
+
+   // Parse command-line options.
+   OptionsParser args(argc, argv);
+   args.AddOption(&mesh_file, "-m", "--mesh",
+                  "Mesh file to use.");
+   args.AddOption(&sltn_file, "-s", "--sltn",
+                  "Solution file to use.");
+   args.AddOption(&ref, "-ref", "--piecewise-linear-ref-factor",
+                  "Scaling factor for resolution of piecewise linear bounds."
+                  " If less than 2, the resolution is picked automatically");
+   args.AddOption(&visualization, "-vis", "--visualization", "-no-vis",
+                  "--no-visualization",
+                  "Enable or disable GLVis visualization.");
+   args.AddOption(&visit, "-visit", "--visit", "-no-visit",
+                  "--no-visit",
+                  "Enable or disable VisIt output.");
+   args.AddOption(&b_type, "-bt", "--basis-type",
+                  "Project input function to a different bases. "
+                  "0 = GL nodes. "
+                  "1 = GLL nodes. "
+                  "2 = uniformly spaced nodes. ");
+   args.AddOption(&continuous, "-h1", "--h1", "-l2", "--l2",
+                  "Use continuous or discontinuous space.");
+   args.AddOption(&nbrute, "-nb", "--nbrute",
+                  "Brute force search for minimum in an array of nxnxn points "
+                  "in each element.");
+   args.Parse();
+   if (!args.Good())
+   {
+      if (myid == 0) { args.PrintUsage(cout); }
+      return 1;
+   }
+   if (myid == 0) { args.PrintOptions(cout); }
+
+   // Initialize and refine the starting mesh.
+   Mesh mesh(mesh_file, 1, 1, false);
+   const int dim = mesh.Dimension();
+   if (continuous)
+   {
+      MFEM_VERIFY(b_type, "Continuous space do not support GL nodes. "
+                  "Please use basis type: 1 for Lagrange interpolants on GLL "
+                  " nodes 2 for positive bases on uniformly spaced nodes.");
+   }
+
+   int *partition = mesh.GeneratePartitioning(nranks);
```

Comment:
```suggestion
   int *partition = mesh.GeneratePartitioning(Mpi::WorldSize());
```
And then can delete the `nranks` variable